core/src/main/java/org/bitcoinj/store/WalletProtobufSerializer.java

```diff
@@ -450,33 +491,40 @@ public Wallet readWallet(NetworkParameters params, @Nullable WalletExtension[] e
             wallet.setDescription(walletProto.getDescription());
         }
 
-        // Read all transactions and insert into the txMap.
-        for (Protos.Transaction txProto : walletProto.getTransactionList()) {
-            readTransaction(txProto, wallet.getParams());
-        }
-
-        // Update transaction outputs to point to inputs that spend them
-        for (Protos.Transaction txProto : walletProto.getTransactionList()) {
-            WalletTransaction wtx = connectTransactionOutputs(txProto);
-            wallet.addWalletTransaction(wtx);
-        }
-
-        // Update the lastBlockSeenHash.
-        if (!walletProto.hasLastSeenBlockHash()) {
+        if (forceReset) {
+            // Should mirror Wallet.reset()
```

Comment:
Mainly because I was unclear on the side effects. `Wallet.reset` also calls [`Wallet.saveLater`](https://github.com/bitcoinj/bitcoinj/blob/master/core/src/main/java/org/bitcoinj/core/Wallet.java#L1403) and [`Wallet.maybeQueueOnWalletChanged`](https://github.com/bitcoinj/bitcoinj/blob/master/core/src/main/java/org/bitcoinj/core/Wallet.java#L2392). I don't know if these calls are safe to make while the Wallet is in the middle of being created.